### PR TITLE
[MAINTENANCE] `invoke deps` task

### PR DESCRIPTION
**Pull request by Kilo59**
Add a new `invoke deps` task.

The simplest use of this task is to synchronize your local core + dev dependencies. This is what happens if you call it with no arguments.

image

The more advanced case is primarily for selecting the appropriate dependencies to install in our CI based on providing a `pytest --marker` arg.

```console
$ invoke deps --help
Usage: inv[oke] [--core-opts] deps [--options] [other tasks here ...]

Docstring:
  Install dependencies for development and testing.

  Specific requirement files needed for a specific test maker can be registered in `MARKER_REQ_MAPPING`,
  `invoke deps` will always check for and use these when installing dependencies.

  If no `markers` or `requirements-dev` are specified, the dev-contrib and
  core requirements are installed.

  Example usage:
  Installing the needed dependencies for running the `external_sqldialect` tests and
  the 'requirements-dev-cloud.txt' dependencies.

  $ invoke deps -m external_sqldialect -r cloud

Options:
  -c, --[no-]constraints   Optional flag to install dependencies with constraints, default True
  -e, --editable-install
  -g, --gx-install
  -m, --markers            Optional marker to install dependencies for. Can be specified multiple times.
  -r, --requirements-dev   Short name of `requirements-dev-*.txt` file to install, e.g. test, spark, cloud etc. Can be specified multiple times.
```

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -777,3 +780,74 @@ def show_automerges(ctx: Context):
...
     else:
         print("\tNo PRs set to automerge")
+
+
+MARKER_REQ_MAPPING: Final[Mapping[str, tuple[str, ...]]] = {
+    "athena": ("reqs/requirements-dev-athena.txt",),
+    "cloud": ("reqs/requirements-dev-cloud.txt",),
+    "pyarrow": ("reqs/requirements-dev-arrow.txt",),
+    "external_sqldialect": ("reqs/requirements-dev-sqlalchemy.txt",),
+}
```
Comment (Kilo59) [inline tasks.py:790]: Might tweak the structure of this mapping to also contain any necessary environmental or service setup needed for the test markers.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -65,8 +65,10 @@ jobs:
         markers:
           - big
           - cli
+          - cloud
           - filesystem
-          - openpyxl or pyarrow or project
+          - openpyxl or pyarrow or project or sqlite
```
Comment (Kilo59) [inline .github/workflows/ci.yml:70]: Also adding these markers to the matrix

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -84,7 +86,9 @@ jobs:
             setup.py
 
       - name: Install dependencies
-        run: pip install -c constraints-dev.txt ".[test]"
+        run: |
+          pip install $(grep -E '^(invoke)' reqs/requirements-dev-contrib.txt)
```
Comment (nicgrayson) [inline .github/workflows/ci.yml:90]: Can this be rolled into the invoke command too?

**Review by nicgrayson — APPROVED** [review]
LGTM

**Pull request merged**